service/worker/batcher/activities.go

```diff
@@ -195,11 +202,12 @@ func (a *activities) getActivityLogger(ctx context.Context) log.Logger {
 	)
 }
 
-func (a *activities) getOperationRPS(rps int) int {
-	if rps <= 0 {
-		return a.rps(a.namespace.String())
+func (a *activities) getOperationRPS(requestedRPS float64) float64 {
+	maxRPS := float64(a.rps(a.namespace.String()))
+	if requestedRPS <= 0 || requestedRPS > maxRPS {
+		return maxRPS
 	}
-	return rps
+	return requestedRPS
```

Comment:
This is the main change; all others are just cosmetic (making fields private, fixing up formatting and comments etc.)